dvc/config.py

```diff
@@ -250,21 +250,24 @@ def load_one(self, level):
         return conf
 
     @staticmethod
-    def _load_paths(conf, filename):
-        abs_conf_dir = os.path.abspath(os.path.dirname(filename))
+    def _resolve(conf_dir, path):
+        from .config_schema import RelPath
 
-        def resolve(path):
-            from .config_schema import RelPath
+        expanded = os.path.expanduser(path)
+        if os.path.isabs(expanded) or re.match(r"\w+://", path):
+            return expanded
 
-            if os.path.isabs(path) or re.match(r"\w+://", path):
-                return path
+        # on windows convert slashes to backslashes
+        # to have path compatible with abs_conf_dir
+        if os.path.sep == "\\" and "/" in path:
+            path = path.replace("/", "\\")
 
-            # on windows convert slashes to backslashes
-            # to have path compatible with abs_conf_dir
-            if os.path.sep == "\\" and "/" in path:
-                path = path.replace("/", "\\")
+        return RelPath(os.path.join(conf_dir, path))
 
-            return RelPath(os.path.join(abs_conf_dir, path))
+    @classmethod
+    def _load_paths(cls, conf, filename):
+        conf_dir = os.path.abspath(os.path.dirname(filename))
+        resolve = partial(cls._resolve, conf_dir)
```

Comment:
Adjusted it to look like `_save_paths` below, and have the same structure here and in tests.